ldap/servers/slapd/abandon.c

```diff
@@ -130,7 +140,8 @@ do_abandon(Slapi_PBlock *pb)
         slapi_log_err(SLAPI_LOG_TRACE, "do_abandon", "op not found\n");
     }
 
-    if (0 == pagedresults_free_one_msgid_nolock(pb_conn, id)) {
+    pthread_mutex_unlock(&(pb_conn->c_mutex));
```

Comment:
Does that mean that one of the possible symptom was a deadlock ?

Reply:
Well everything is possible after an heap corruption but deadlock is really very improbable, because the previous code was not locking at all (causing the race condition and the heap corruption) 
But my first idea was to simply protect the pagedresults_free_one_msgid_nolock with the paged mutex lock and I realized that this could generate deadlock if the connection mutex was not freed before.